src/Widgets/WindowClone.vala

```diff
@@ -295,53 +295,151 @@ namespace Gala {
             var offset_x = monitor_geom.x;
             var offset_y = monitor_geom.y;
 
-            save_easing_state ();
-            set_easing_mode (MultitaskingView.ANIMATION_MODE);
-            set_easing_duration (animate ? MultitaskingView.ANIMATION_DURATION : 0);
+            var initial_x = this.x;
+            var initial_y = this.y;
+            var initial_width = this.width;
+            var initial_height = this.height;
 
-            set_position (outer_rect.x - offset_x, outer_rect.y - offset_y);
-            set_size (outer_rect.width, outer_rect.height);
+            var target_x = outer_rect.x - offset_x;
+            var target_y = outer_rect.y - offset_y;
 
-            if (should_fade ())
-                opacity = 0;
+            GestureAnimationDirector.OnBegin on_animation_begin = () => {
+                window_icon.set_easing_duration (0);
+            };
 
-            restore_easing_state ();
+            GestureAnimationDirector.OnUpdate on_animation_update = (percentage) => {
```

Comment:
There's a lot of code repetition in these lambdas. Perhaps there's a way to put these in a common function?

Reply:
I moved the animation value calculation to a common method, but I don't think I can move more code to a common place because each lambda needs to animate different properties in different objects. 
Could you provide an example, please?